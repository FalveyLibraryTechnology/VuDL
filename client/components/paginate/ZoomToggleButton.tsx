import React from "react";

interface ZoomToggleButtonProps {
    toggleZoom: React.MouseEventHandler<HTMLButtonElement>;
    zoom: boolean;
}

const ZoomToggleButton = ({ toggleZoom, zoom }: ZoomToggleButtonProps): React.ReactElement => {
    return <button onClick={toggleZoom}>{`Turn Zoom ${zoom ? "Off" : "On"}`}</button>;
};

export default ZoomToggleButton;
