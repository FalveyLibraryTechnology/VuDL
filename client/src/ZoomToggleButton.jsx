import React from "react";
import PropTypes from "prop-types";

const ZoomToggleButton = ({ paginator }) => {
    return <button onClick={paginator.toggleZoom}>{`Turn Zoom ${paginator.state.zoom ? "Off" : "On"}`}</button>;
};

ZoomToggleButton.propTypes = {
    // JobPaginator
    paginator: PropTypes.shape({
        toggleZoom: PropTypes.func,
        state: PropTypes.shape({
            zoom: PropTypes.bool,
        }),
    }),
};

export default ZoomToggleButton;
