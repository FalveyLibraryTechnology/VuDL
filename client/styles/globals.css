:root,
[color-scheme="light"] {
    --body-bg: #fafbfc;
    --heading-color: #4d4d4d;
    --link-color: #008aed;

    --btn-bg: #485a6c;
    --btn-text: #ffffff;
    --btn-border: #275575;
    --btn-focus-bg: #3498db;

    --btn-primary-bg: #27ae60;
    --btn-primary-text: #ffffff;
    --btn-primary-border: #275575;
    --btn-primary-focus-bg: #155724;

    --btn-danger-bg: red;
    --btn-danger-text: #ffffff;
    --btn-danger-border: #a10705;
    --btn-danger-focus-bg: #a10705;

    --btn-logout-bg: #1abc9c;
    --btn-logout-text: #ffffff;
    --btn-logout-border: #16a085;
    --btn-logout-focus-bg: #1abc9c;

    --btn-toggle-bg: #7f8c8d;
    --btn-toggle-text: #ffffff;
    --btn-toggle-border: #34495e;
    --btn-toggle-focus-bg: #53626f;

    --group-border: #53626f;

    --input-border: #6a737d;
    --input-focus-border: #2188ff;

    --label-hover: #c8e1ff;

    --page-list-bg: #ffffff;
    --page-list-border: #c1c1c1;

    --thumbnail-bg: #e1e1e1;
    --thumbnail-focus-bg: #51aded;
    --thumbnail-number-bg: #ffffff;
    --thumbnail-number-text: #000000;
    --thumbnail-preview-border: #000000;

    --tree-item-focus-bg: #000000;
    --tree-item-focus-text: #ffffff;

    --shadow-gray: #c8e1ff;
}

[color-scheme="dark"] {
}
