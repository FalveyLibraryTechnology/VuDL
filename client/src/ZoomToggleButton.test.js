import React from "react";
import { beforeEach, describe, expect, it, jest } from "@jest/globals";
import { shallow, render, mount } from "enzyme";
import toJson from "enzyme-to-json";
import ZoomToggleButton from "./ZoomToggleButton";

describe("ZoomToggleButton", () => {
    let props;

    beforeEach(() => {
        props = {
            paginator: {
                toggleZoom: jest.fn(),
                state: {
                    zoom: false,
                },
            },
        };
    });

    it("renders", () => {
        const wrapper = shallow(<ZoomToggleButton {...props} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders correctly when zoom is off", () => {
        const wrapper = render(<ZoomToggleButton {...props} />);
        expect(wrapper.text().includes("Turn Zoom On")).toBeTruthy();
    });

    it("renders correctly when zoom is on", () => {
        props.paginator.state.zoom = true;
        const wrapper = render(<ZoomToggleButton {...props} />);
        expect(wrapper.text().includes("Turn Zoom Off")).toBeTruthy();
    });

    it("calls toggle zoom when button is clicked", () => {
        props.paginator.state.zoom = true;
        const wrapper = mount(<ZoomToggleButton {...props} />);

        expect(props.paginator.toggleZoom).not.toHaveBeenCalled();

        wrapper.find("button").simulate("click");

        expect(props.paginator.toggleZoom).toHaveBeenCalled();
    });
});
