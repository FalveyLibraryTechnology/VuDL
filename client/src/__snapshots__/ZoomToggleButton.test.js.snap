// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZoomToggleButton renders 1`] = `
<button
  onClick={[MockFunction]}
>
  Turn Zoom On
</button>
`;
